R/utils-ui.R

```diff
@@ -196,36 +196,31 @@ message <- function(...) {
     instead of {.fun message}")
 }
 
-warn_for_verbose <- function(verbose = TRUE, env = caller_env()) {
-  # this is about whether `verbose` was present in the **user's** call to the
-  # calling function
-  # don't worry about the `verbose = TRUE` default here
+warn_for_verbose <- function(verbose = TRUE,
+                             env = caller_env(),
+                             user_env = caller_env(2)) {
+  # This function is not meant to be called directly, so don't worry about its
+  # default of `verbose = TRUE`.
+  # In authentic, indirect usage of this helper, this picks up on whether
+  # `verbose` was present in the **user's** call to the calling function.
   if (!lifecycle::is_present(verbose) || isTRUE(verbose)) {
     return(invisible())
   }
 
-  called_from <- sys.parent(1)
-  if (called_from == 0) {
-    # called from global env, presumably in a test or during development
-    caller <- "some_googledrive_function"
-  } else {
-    called_as <- sys.call(called_from)
-    if (is.call(called_as) && is.symbol(called_as[[1]])) {
-      caller <- as.character(called_as[[1]])
-    } else {
-      caller <- "some_googledrive_function"
-    }
-  }
   lifecycle::deprecate_warn(
     when = "2.0.0",
-    what = glue("{caller}(verbose)"),
+    what = I("The `verbose` argument"),
```

Comment:
Why this change? Is it to simplify the pattern for the book?

I think this is where the mini-DSL for specifying deprecated features falls apart, i.e. it's nice for direct usage but causes complications when programmed.

Ideally we'd keep mentioning the function where the argument is deprecated. And ideally we wouldn't have to specify it and automatically infer it from `env`.

So I guess to help programmatic usage we'd need some sort of ctor like `what = deprec(arg = "my_arg")`. You could supply `fn = ` but it would be automatically inferred from `env` if not supplied.

Reply:
> Why this change? Is it to simplify the pattern for the book?

Yes, exactly.